src/lib.rs

```diff
@@ -626,6 +626,17 @@ arbitrary_array! { 32, (T, a) (T, b) (T, c) (T, d) (T, e) (T, f) (T, g) (T, h)
 (T, z) (T, aa) (T, ab) (T, ac) (T, ad) (T, ae) (T, af)
 (T, ag) }
 
+impl<'a> Arbitrary<'a> for &'a [u8] {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
+        u.get_bytes(u.len())
+    }
+
+    #[inline]
+    fn size_hint(depth: usize) -> (usize, Option<usize>) {
+        crate::size_hint::and(<usize as Arbitrary>::size_hint(depth), (0, None))
```

Comment:
I cargo culted this line from the `Vec<A>` implementation. Can someone explain why we defer to the `usize` `Arbitrary` implementation? My initial idea was to just return `(0, None)` here, since we can construct a slice of bytes from nothing

Reply:
a4bbf9c1 added the method/impl to `Vec` cc @fitzgen

This impl effectively means `(8, None)` is always produced. Given the documentation on the `size_hint` method, I imagine `Arbitrary for Vec` would generate a `usize` length first, and then the number of elements or something along those lines. As thus the implementation needed at least 8 bytes of unstructured input to operate.

Reply:
Yeah the `usize` size hint is for the vec's length.

Reply:
Thanks for clarifying! I think I understand the `usize` part now. And why do we use `size_hint::and(...)` here? Isn't this identical to just `<usize as Arbitrary>::size_hint(depth)`?

Reply:
Yes it is. I'm not sure if I originally wrote this line or not, but in general I stylistically use `crate::size_hint::and(..)` even when its a no-op for nested fields or nested arbitrary calls just as a future proofing mechanism in case they grow in size sometime in the future or something. That doesn't really apply to this call, since it is using a hard coded value, rather than a nested `size_hint` call, so it should be safe to remove it.

Reply:
https://github.com/rust-fuzz/arbitrary/pull/67/commits/7d28b784114119f1b7a64f17e9c8361eb729905f